code/Helper/Entity/Categoryhelper.php

```diff
@@ -118,6 +118,8 @@ public function getAllAttributes()
 
     public function getObject(Mage_Catalog_Model_Category $category)
     {
+        $category->load($category->getId());
```

Comment:
I fixed the image field not being available when doing $category->getImageUrl() by adding the attribute 'image' in getCategoryCollectionQuery(). This way you can get the image without having to load each category of the collection again.